llvm-bazel/.bazelrc

```diff
@@ -62,6 +62,48 @@ build:generic_gcc --copt=-Wno-misleading-indentation --host_copt=-Wno-misleading
 # Use `-Werror` for GCC to make sure warnings don't slip past.
 build:generic_gcc --copt=-Werror --host_copt=-Werror
 
+###############################################################################
+# Windows specific flags for building with VC.
+###############################################################################
+
+build:msvc --copt=/WX     # Treat warnings as errors...
+
+# ...but disable the ones that are violated
+build:msvc --copt=/wd4624 # destructor was implicitly defined as deleted
+build:msvc --copt=/wd4244 # possible loss of data
+build:msvc --copt=/wd4005 # macro redefinition
+build:msvc --copt=/wd4141 # inline used more than once
+build:msvc --copt=/wd4267 # initializing: possible loss of data
+build:msvc --copt=/wd4273 # multiple definitions with different dllimport
+build:msvc --copt=/wd4319 # zero-extending after complement
+build:msvc --copt=/wd4804 # comparisons between bool and int
+build:msvc --copt=/wd4805 # comparisons between bool and int
+
+build:msvc --linkopt=/WX # Treat warnings as errors...
+
+# ...but disable the ones that are violated.
+build:msvc --linkopt=/IGNORE:4001 # no object files
+build:msvc --linkopt=/IGNORE:4217 # mismatch import/export declspec
```

Comment:
I would be interested to know why these two need to be disabled, especially the last one seems surprising to me.

Reply:
4001 is from having `cc_binary` with only deps and no srcs, I think. We get errors like:

> ERROR: C:/users/kbuilder/_bazel_kbuilder/paxle3de/external/llvm-project/llvm/BUILD:471:10: Linking of rule '@llvm-project//llvm:llvm-tblgen' failed (Exit 4001)
LINK : warning LNK4001: no object files specified; libraries used

4217 is not showing up for me now... Hmm I must have copy-pasta'd this from iree and then gotten it mixed up with 4273 when I was trimming things.

Reply:
Okay. 4001 sounds benign, but 4217 seems a bit scary so dropping that is probably good.